internal/server/server.go

```diff
@@ -285,12 +308,27 @@ func (s *AsyncServer) executeCommandToBuffer(redisCmd *cmd.RedisCmd, buf *bytes.
 	}
 
 	resp := <-s.ioChan
-	if resp.EvalResponse.Error != nil {
-		buf.WriteString(resp.EvalResponse.Error.Error())
+
+	val, ok := WorkerCmdsMeta[redisCmd.Cmd]
+
+	// TODO: Remove this conditional check and if (true) condition when all commands are migrated
+	if !ok {
+		buf.Write(resp.EvalResponse.Result.([]byte))
+	} else {
+
+		if val.CmdType == Global {
+			buf.Write(val.RespNoShards(redisCmd.Args))
+			return
+		}
+
+		if resp.EvalResponse.Error != nil {
+			handleMigratedResp(resp.EvalResponse.Error, buf)
```

Comment:
There should be a return after this. Else `handleMigratedResp(resp.EvalResponse.Result, buf)` would also get called.